module/Database/src/Mapper/Member.php

```diff
@@ -120,7 +120,7 @@ public function find(int $lidnr): ?MemberModel
             ->where('m.lidnr = :lidnr')
             ->leftJoin('m.installations', 'r')
             ->leftJoin('m.lists', 'l')
-            ->andWhere('(r.function = \'Lid\' OR r.function IS NULL)');
+            ->andWhere('(r.function = \'Lid\' OR r.function = \'Inactief Lid\' OR r.function IS NULL)');
```

Comment:
I couldn't really add this anywhere, but I think we also should add a few lines to the CheckerService since it is allowed to be installed as an inactive member and not as an active member. 

While we are at it, me actually want to ensure we don't have "Lid" when we have "Inactief Lid"

```diff
public function checkMembersHaveRoleButNotInOrgan(MeetingModel $meeting): array
    {
        $errors = [];
        $membersArray = $this->installationService->getCurrentRolesPerOrgan($meeting);

        foreach ($membersArray as $organsMembers) {
            foreach ($organsMembers as $memberRoles) {
                if (!isset($memberRoles['Lid'])) {
                    foreach ($memberRoles as $role => $installation) {
+                       if ('Inactief Lid' === $role) {
+                           continue;
+                       }
                        $errors[] = new Error\MemberHasRoleButNotInOrgan($meeting, $installation, $role);
                    }
+               } elseif (isset($memberRoles['Inactief Lid'])) {
+                   $errors[] = new Error\MemberIsInactiveInOrganButHasOtherRole($meeting, $installation, $role);
                }
            }
        }
        return $errors;
    }

```

Reply:
I think we want more types of errors in this regard. An error if a member is active and inactive (perhaps a further distinction regarding the number of roles to determine if they should be active or inactive*), an error if a member is inactive but still has a role, and an error if a member has a role but is not active (nor inactive).

```diff
     /**
-     * Checks if members still have a role in an organ (e.g. they are treasurer)
-     * but they are not a member of the organ anymore
+     * Checks if members still have a role in an organ (e.g. they are treasurer) but they are not a member of the organ
+     * anymore, or they are an inactive member.
      *
      * @param MeetingModel $meeting After which meeting do we do the validation
      *
-     * @return array Array of errors that may have occured.
+     * @return array Array of errors that may have occurred.
      */
-    public function checkMembersHaveRoleButNotInOrgan(MeetingModel $meeting): array
+    public function checkMembersHaveRolesButInactiveOrNotInOrgan(MeetingModel $meeting): array
     {
         $errors = [];
-        $membersArray = $this->installationService->getCurrentRolesPerOrgan($meeting);
+        $organs = $this->installationService->getCurrentRolesPerOrgan($meeting);

-        foreach ($membersArray as $organsMembers) {
-            foreach ($organsMembers as $memberRoles) {
-                if (!isset($memberRoles['Lid'])) {
+        foreach ($organs as $organMembers) {
+            foreach ($organMembers as $memberRoles) {
+                if (
+                    isset($memberRoles['Lid'])
+                    && isset($memberRoles['Inactief Lid'])
+                ) {
+                    // Member is active AND inactive in the same organ.
+                    $errors[] = new Error\MemberActiveAndInactiveInOrgan($meeting, /** unclear which installation should be used (maybe both?) */);
+                } elseif (
+                    !isset($memberRoles['Lid'])
+                    && isset($memberRoles['Inactief Lid'])
+                    && count($memberRoles) > 1
+                ) {
+                    // Member is inactive but still has roles.
+                    foreach ($memberRoles as $role => $installation) {
+                        $errors[] = new Error\MemberIsInactiveInOrganButHasOtherRole($meeting, $installation, $role);
+                    }
+                } elseif (
+                    !isset($memberRoles['Lid'])
+                    && !isset($memberRoles['Inactief Lid'])
+                ) {
+                    // Member is not active (nor inactive) but still has roles.
                     foreach ($memberRoles as $role => $installation) {
                         $errors[] = new Error\MemberHasRoleButNotInOrgan($meeting, $installation, $role);
                     }
                 }
             }
         }
+
         return $errors;
     }
```

*: With additional roles assume that inactive state is incorrect. If there are no additional roles assume active state is incorrect. Like this:

```php
// ...
if (
    isset($memberRoles['Lid'])
    && isset($memberRoles['Inactief Lid'])
) {
    // Member is active AND inactive in the same organ.
    if (count($memberRoles > 2)) {
        $errors[] = new Error\MemberActiveWithRolesAndInactiveInOrgan($meeting, $memberRoles['Inactief Lid']);
    } else {
        $errors[] = new Error\MemberActiveAndInactiveInOrgan($meeting, $memberRoles['Lid']);
    }
} elseif (
    !isset($memberRoles['Lid'])
    && isset($memberRoles['Inactief Lid'])
    && count($memberRoles) > 1
) {
// ...
```